dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/deduplication/DeduplicationControllerTest.java

```diff
@@ -177,73 +176,137 @@ void shouldGetNonPaginatedItemsWithPagingSetToFalse() {
   }
 
   @Test
-  void shouldThrowPostPotentialDuplicateWhenMissingDuplicateTeiInPayload() throws Exception {
+  void shouldThrowBadRequestExceptionWhenMissingDuplicateTeInPayload() throws Exception {
     PotentialDuplicate potentialDuplicate = new PotentialDuplicate(UID.of(origin), null);
-    assertStatus(
-        HttpStatus.BAD_REQUEST,
-        POST(ENDPOINT, objectMapper.writeValueAsString(potentialDuplicate)));
+
+    POST(ENDPOINT, objectMapper.writeValueAsString(potentialDuplicate))
+        .content(HttpStatus.BAD_REQUEST);
+  }
+
+  @Test
+  void shouldThrowNotFoundExceptionWhenTeInPayloadDoesNotExist() throws Exception {
+    String nonExistentUID = CodeGenerator.generateUid();
```

Comment:
```suggestion
    String nonExistentUID = UID.generate();
```

makes the same call under the hood but is nicer to read 😁